Formula/d/dwarfs.rb

```diff
@@ -0,0 +1,167 @@
+class Dwarfs < Formula
+  desc "Fast high compression read-only file system for Linux, Windows, and macOS"
+  homepage "https://github.com/mhx/dwarfs"
+  url "https://github.com/mhx/dwarfs/releases/download/v0.10.1/dwarfs-0.10.1.tar.xz"
+  sha256 "db785e0e0f257fa4363d90153db34127add4552791a72998b30ded787840d039"
+  license "GPL-3.0-or-later"
+
+  # At least Apple Clang 15 is required as we need several C++20 features.
+  NEED_LLVM_CLANG = (OS.mac? && DevelopmentTools.clang_build_version < 1500).freeze
```

Comment:
I don't think we want to define constants in the formulae for this. I don't think we have any precedent for it.

Reply:
So we'd rather keep repeating the same statement over and over again and violate DRY and risk individual instances becoming inconsistent?

Do you have an idea how else to avoid the redundancy?

Reply:
We don't avoid the redundancy, we don't care about DRY. We want to be explicit in the configuration because the readability of the formula is the most important thing.

Reply:
Interesting. I'd personally consider something like `NEED_LLVM_CLANG` to be more readable. ¯\_(ツ)_/¯